File: Tugas12/Anggota.php
<?php
//Simpanlah dengan nama file : Anggota.php
require_once 'database.php';
class Anggota 
{
    private $db;
    private $table = 'anggota';
    public $kode_anggota = "";
    public $nama = "";
    public $jk = "";
    public $prodi = "";
    public $Alamat = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_anggota(int $kode_anggota)
    {
        $query = "SELECT * FROM $this->table WHERE kode_anggota = $kode_anggota";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_anggota`,`nama`,`jk`,`prodi`,`Alamat`) VALUES ('$this->kode_anggota','$this->nama','$this->jk','$this->prodi','$this->Alamat')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_anggota = '$this->kode_anggota', nama = '$this->nama', jk = '$this->jk', prodi = '$this->prodi', Alamat = '$this->Alamat' 
        WHERE idanggota = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_anggota($kode_anggota): int
    {
        $query = "UPDATE $this->table SET kode_anggota = '$this->kode_anggota', nama = '$this->nama', jk = '$this->jk', prodi = '$this->prodi', Alamat = '$this->Alamat' 
        WHERE kode_anggota = $kode_anggota";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE idanggota = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_anggota($kode_anggota): int
    {
        $query = "DELETE FROM $this->table WHERE kode_anggota = $kode_anggota";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Tugas12/Anggota.py
import requests
import json
class Anggota:
    def __init__(self):
        self.__id=None
        self.__kode_anggota = None
        self.__nama = None
        self.__jk = None
        self.__prodi = None
        self.__Alamat = None
        self.__url = "http://localhost/perpus/anggota_api.php"
                    
    @property
    def id(self):
        return self.__id
    
    @property
    def kode_anggota(self):
        return self.__kode_anggota
        
    @kode_anggota.setter
    def kode_anggota(self, value):
        self.__kode_anggota = value
    
    @property
    def nama(self):
        return self.__nama
        
    @nama.setter
    def nama(self, value):
        self.__nama = value
   
    @property
    def jk(self):
        return self.__jk
        
    @jk.setter
    def jk(self, value):
        self.__jk = value
    
    @property
    def prodi(self):
        return self.__prodi
        
    @prodi.setter
    def prodi(self, value):
        self.__prodi = value
    @property
    def Alamat(self):
        return self.__Alamat
        
    @Alamat.setter
    def Alamat(self, value):
        self.__Alamat = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_anggota(self, kode_anggota):
        url = self.__url+"?kode_anggota="+kode_anggota
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_anggota']
            self.__kode_anggota = item['kode_anggota']
            self.__nama = item['nama']
            self.__jk = item['jk']
            self.__prodi = item['prodi']
            self.__Alamat = item['Alamat']
        return data
    def simpan(self):
        payload = {
            "kode_anggota":self.__kode_anggota,
            "nama":self.__nama,
            "jk":self.__jk,
            "prodi":self.__prodi,
            "Alamat":self.__Alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_anggota(self, kode_anggota):
        url = self.__url+"?kode_anggota="+kode_anggota
        payload = {
            "kode_anggota":self.__kode_anggota,
            "nama":self.__nama,
            "jk":self.__jk,
            "prodi":self.__prodi,
            "Alamat":self.__Alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_anggota(self,kode_anggota):
        url = self.__url+"?kode_anggota="+kode_anggota
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text